sources/models/src/modeled_types/shared.rs

```diff
@@ -250,3 +252,100 @@ mod test_url {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// FriendlyVersion represents a version string that can optionally be prefixed with 'v'.
+/// It can also be set to 'latest' to represent the latest version. It stores the original string
+/// and makes it accessible through standard traits.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct FriendlyVersion {
+    inner: String,
+}
+
+impl TryFrom<&str> for FriendlyVersion {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        if input == "latest" {
+            return Ok(FriendlyVersion {
+                inner: input.to_string(),
+            });
+        }
+        // If the string begins with a 'v', skip it before checking if it is valid semver.
+        let version = if input.starts_with('v') {
+            &input[1..]
+        } else {
+            input
+        };
+
+        if version.parse::<semver::Version>().is_ok() {
+            return Ok(FriendlyVersion {
+                inner: input.to_string(),
+            });
+        }
+        error::InvalidVersion { input }.fail()
+    }
+}
+
+impl TryFrom<FriendlyVersion> for semver::Version {
+    type Error = semver::SemVerError;
+
+    fn try_from(input: FriendlyVersion) -> Result<semver::Version, Self::Error> {
+        // If the string begins with a 'v', skip it before conversion
+        let version = if input.inner.starts_with('v') {
+            &input.inner[1..]
+        } else {
+            &input.inner
+        };
+        Version::from_str(version)
+    }
+}
+
+string_impls_for!(FriendlyVersion, "FriendlyVersion");
+
+#[cfg(test)]
+mod test_version {
+    use super::FriendlyVersion;
+    use semver::{SemVerError, Version};
+    use std::convert::TryFrom;
+    use std::convert::TryInto;
+
+    #[test]
+    fn good_version_strings() {
+        for ok in &[
+            "1.0.0",
+            "v1.0.0",
+            "1.0.1-alpha",
+            "v1.0.1-alpha",
+            "1.0.2-alpha+1.0",
+            "v1.0.2-alpha+1.0",
+            "1.0.3-beta.1.01",
+            "v1.0.3-beta.1.01",
+            "1.1.0-rc.1.1",
+            "v1.1.0-rc.1.1",
+            "latest",
+        ] {
+            FriendlyVersion::try_from(*ok).unwrap();
+            // Test conversion to semver::Version
+            if *ok != "latest" {
+                let _: Version = FriendlyVersion {
```

Comment:
nit: You might be able to do a similar change here to keep things consistent, `Version::try_from(FriendlyVersion { bla })` rather than bringing in TryInto.